testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_replicator_encryption_hook.py

```diff
@@ -514,6 +517,97 @@ def test_encryption_with_two_dbs(params_from_base_test_setup):
     assert deep_dict_compare(encrypted_doc['encrypted_field_dict']['value'], dict)
 
 
+@pytest.mark.listener
+@pytest.mark.callback
+@pytest.mark.sanity
+def test_replication_complex_doc_encryption(params_from_base_test_setup):
+    """
+    @summary:
+    Create various types of Encrypted values.
+    https://docs.google.com/spreadsheets/d/1-E7qv8UlR3-AyhaKvagsFtwB-i4JOBZWGKHJn6JDSqY/edit#gid=0
+    test 13
+
+    1. Have SG and CBL up and running
+    2. Create a simple document with encryption property (String, int, Double, Float, dict , Array, and Dict)
+    3. Start the replicator and make sure documents are
+    replicated on SG. Verify encrypted fields and Verify data is encrypted.
+    4.  Verify Updates on the Encrypted doc on CBL
+    5.  Start the replicator and verify the docs
+    6. Verify Delete on the Encrypted document on CBL and replicate
+    """
+
+    liteserv_platform = params_from_base_test_setup["liteserv_platform"]
+    if "c-" not in liteserv_platform.lower():
+        pytest.skip('This test cannot run other than C.')
+
+    sg_db = "db"
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_config = params_from_base_test_setup["sg_config"]
+    db = params_from_base_test_setup["db"]
+    cbl_db = params_from_base_test_setup["source_db"]
+    sync_gateway_version = params_from_base_test_setup["sync_gateway_version"]
+    number_of_updates = 2
+    sg_client = MobileRestClient()
+
+    if sync_gateway_version < "2.0.0":
+        pytest.skip('This test cannot run with sg version below 2.0')
+    channels_sg = ["ABCD"]
+    username = "autotest"
+    password = "password"
+
+    c = cluster.Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    # Create docs in CBL
+    encryptable = ReplicatorCallback(base_url)
+    documentObj = Document(base_url)
+    dictionary = Dictionary(base_url)
+
+    # Create encrypted value
```

Comment:
Can you make sure you follow the format to match the comments with steps to have better readability and understand the test